server/src/test/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregationBuilderTests.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.bucket.histogram;
+
+import org.elasticsearch.test.ESTestCase;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class AutoDateHistogramAggregationBuilderTests extends ESTestCase {
+
+    public void testInvalidInterval() {
+        AutoDateHistogramAggregationBuilder builder = new AutoDateHistogramAggregationBuilder("name");
+        IllegalArgumentException wrongIntervalName = expectThrows(IllegalArgumentException.class,
+            () -> builder.setMinimumIntervalExpression("foobar"));
+        assertTrue(wrongIntervalName.getMessage().startsWith("minimum_interval must be one of"));
+    }
+
+    public void testBuildRoundingsWithNullParameter() {
+        AutoDateHistogramAggregationBuilder.RoundingInfo[] roundings = AutoDateHistogramAggregationBuilder.buildRoundings(null, null);
+        assertThat(roundings.length, equalTo(6));
+    }
+
+    public void testBuildRoundingsWithMinInterval() {
+        AutoDateHistogramAggregationBuilder.RoundingInfo[] roundings = AutoDateHistogramAggregationBuilder.buildRoundings(null, "year");
+        assertThat(roundings.length, equalTo(1));
+        roundings = AutoDateHistogramAggregationBuilder.buildRoundings(null, "second");
+        assertThat(roundings.length, equalTo(6));
+    }
+
+}
```

Comment:
Can we make these tests check the roundings that are created to make sure they contain the correct rounding levels, not just the number of roundings returned?